yt_dlp/extractor/twitch.py

```diff
@@ -191,6 +190,24 @@ def _get_thumbnails(self, thumbnail):
             'url': thumbnail,
         }] if thumbnail else None
 
+    def _extract_twitch_m3u8_formats(self, type, id, token, signature):
+        URLS_MAP = {
+            'video': '%s/vod/%s.m3u8',
+            'stream': '%s/api/channel/hls/%s.m3u8'
+        }
+        query = {
+            'allow_source': 'true',
+            'allow_audio_only': 'true',
+            'allow_spectre': 'true',
+            'p': random.randint(1000000, 10000000),
+            'player': 'twitchweb',
+            'playlist_include_framerate': 'true',
+            'sig': signature,
+            'token': token,
+        }
+        return self._extract_m3u8_formats(
+            URLS_MAP[type] % (self._USHER_BASE, id), id, 'mp4', query=query)
```

Comment:
Maybe it would be simpler to do it like this:
```suggestion
    def _extract_twitch_m3u8_formats(self, video_id, token, signature):
        """Subclasses must define _M3U_PATH"""
        return self._extract_m3u8_formats(
            f'{self._USHER_BASE}/{self._M3U8_PATH}/{video_id}.m3u8', video_id, 'mp4', query={
                'allow_source': 'true',
                'allow_audio_only': 'true',
                'allow_spectre': 'true',
                'p': random.randint(1000000, 10000000),
                'player': 'twitchweb',
                'playlist_include_framerate': 'true',
                'sig': signature,
                'token': token,
            })
```
and then in the respective concrete IE classes:
```py
    _M3U8_PATH = 'vod'
```
```py
    _M3U8_PATH = 'api/channel/hls'
```
(and remove the `'video'` and `'stream'` args from the method calls)

Reply:
Why not just pass the m3u8 path itself as an argument? Each case is only used once

Reply:
https://github.com/yt-dlp/yt-dlp/pull/8968/commits/e8715c215cbdc3c7dc71ca71db0743ad578e3990 (#8968)